apache/superset
# feat(helm): optionally set db uri from env

### SUMMARY

Allow one to set an alternate sqlalchemy database uri (e.g. - different SQL dialect, mTLS, etc.) via environment variable.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


### TESTING INSTRUCTIONS

These commands should yield equivalent results:

```shell
helm template \
  supserset helm/superset |
  kubectl apply -f-
```

```shell
helm template \
  --set supersetNode.connections.db_user="" \
  --set supersetNode.connections.db_pass="" \
  --set supersetNode.connections.db_name="" \
  --set extraSecretEnv.SQLALCHEMY_DATABASE_URI=postgresql+psycopg2://superset:superset@superset-postgresql:5432/superset \
  superset helm/superset |
  kubectl apply -f-
```

N.B. - supersetNode connections host & port are not empty strings in the subsequent command so that the TCP test in the init container passes, but nullifying user, pass & name demonstrates that the SQLALCHEMY_DATABASE_URI environment variable overrides the default behavior.

### ADDITIONAL INFORMATION
